Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandPartyJoin.java

```diff
@@ -0,0 +1,56 @@
+package com.universeprojects.miniup.server.commands;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.universeprojects.cacheddatastore.CachedDatastoreService;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.miniup.server.commands.framework.Command;
+import com.universeprojects.miniup.server.commands.framework.UserErrorMessage;
+
+/**
+ * 
+ * 
+ * @author dylan
+ * 
+ * Allows a player to join a party if they are already not apart of a party, not in combat, the party is accepting members, or already apart of the same party,
+ *
+ */
+public class CommandPartyJoin extends Command {
+
+	public CommandPartyJoin(ODPDBAccess db, HttpServletRequest request,
+			HttpServletResponse response) {
+		super(db, request, response);
+	}
+
+	@Override
+	public void run(Map<String, String> parameters) throws UserErrorMessage {
+		ODPDBAccess db = getDB();
+		CachedDatastoreService ds = getDS();
+
+		CachedEntity character = db.getCurrentCharacter();
+		String toJoin = parameters.get("partyCode");
+		String currentParty = (String) character.getProperty("partyCode");
+		
+		if (toJoin.equals(currentParty)) {
+			throw new UserErrorMessage("You are already apart of this party!");
+		}
+		if (currentParty != null && !currentParty.trim().equals("")) {
+			throw new UserErrorMessage("You are already in a party! You must leave this one first!");
+		}
+		
+		CachedEntity leader = db.getPartyLeader(ds, toJoin);
+		if (leader.getProperty("partyJoinsAllowed").equals("FALSE")) {
+			throw new UserErrorMessage("This party is not accepting members currently!");
+		}
+		
+		if (character.getProperty("mode").equals("COMBAT")) {
+			throw new UserErrorMessage("You cannot join a party while in combat!");
+		}
+		
+		db.doJoinParty(ds, character, toJoin);
```

Comment:
I think it would be better to just collapse this call into the command. I can't think of any reason why we would want to call doJoinparty() from anywhere else.